sources: add Telegram source (#15749)
goauthentik/authentik: website/docs/users-sources/sources/social-logins/telegram/index.md

---
title: Telegram
support_level: community
---

> Allows users to authenticate using their Telegram account via Telegram application

Reviewer: ```suggestion
Configuring Telegram as a source allows users to authenticate using their Telegram account credentials within authentik.
```

## Telegram

> To use Telegram as a source, you need to register a Telegram bot first:

Reviewer: ```suggestion
To use Telegram as a source, you first need to register a Telegram bot:
```

1. Start a chat with `@BotFather` on Telegram.
2. Use `/newbot` command to create a new bot. Pick a username for your new bot (e.g., `my_bot`).
   Note the username you've chosen - you'll need it when setting up the source in authentik.
3. BotFather will provide you with a token for your bot. Make a record of the token as well.
4. Link the bot to your authentik domain name using `/setdomain` command.
   Please note that **the domain name must exactly match the FQDN of the authentik installation**.

Now as the bot is configured you can proceed to creating a source in authentik.

## authentik

> In authentik, go to _Directory_ > _Federation & Social login_ and click _Create_.

Reviewer: For UI elements we use bold. Italics are only used for emphasis.

Select _Telegram source_ and configure your source:

> - _Name:_ Choose a name

Reviewer: ```suggestion
- **Name**: Choose a name
```

> - _Slug:_ Set a slug

Reviewer: ```suggestion
- **Slug**: Set a slug
```

- _Bot username:_ Set the username of your Telegram bot (e.g., `my_bot`)
- _Bot token:_ Set the token of your Telegram bot
- _"Request access to send messages from your bot":_ enable this if you need your bot to be able to
  send messages to the users who logged in to authentik with it.

Save, and you now have Telegram as a source.

:::note
For instructions on how to display the new source on the authentik login page, refer to the [Add sources to default login page documentation](../../index.md#add-sources-to-default-login-page).
:::

## Telegram source property mappings

See the [overview](../../property-mappings/index.md) for information on how property mappings work.

### Expression data

The following variables are available to OAuth source property mappings:

- `info`: A Python dictionary containing Telegram user data:
    - `id` - Telegram user ID

>     - `username` - Username of the user. May not be present

Reviewer: ```suggestion
    - `username` - Username of the user. Might not be present.
```

>     - `first_name` - First name of the user. May not be present

Reviewer: ```suggestion
    - `first_name` - First name of the user. Might not be present.
```

>     - `last_name` - Last name of the user. May not be present

Reviewer: ```suggestion
    - `last_name` - Last name of the user. Might not be present.
```
